pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -238,132 +257,191 @@ func TestReconcile(t *testing.T) {
 		PipelineResources: []*v1alpha1.PipelineResource{gitResource, anotherGitResource, imageResource},
 	}
 	for _, tc := range []struct {
-		name          string
-		taskRun       *v1alpha1.TaskRun
-		wantBuildSpec buildv1alpha1.BuildSpec
+		name      string
+		taskRun   *v1alpha1.TaskRun
+		wantBuild *buildv1alpha1.Build
```

Comment:
This is the reason for most of the changes in this file. I could isolate the specific test I added to avoid most of the changes if desired, or go even farther and move to using a Pod builder as discussed in the TODO I added:

> // TODO: Using MakePod means that a diff will not catch issues
> // specific to the Build to Pod translation (e.g. if labels are
> // not propagated in MakePod). To avoid this issue we should create
> // a builder for Pods and use that instead.

Any thoughts/feedback would be appreciated.

Reply:
oh, I made a similar change in pr #491 to not use `MakePod` and just put down the expected Containers specs which is a bigger change, but I had to validate the container's names

Reply:
Yeah that's exactly what I had in mind, these 2 PRs will definitely have a ton of conflicts with each other in that file. I think it would be good to go with your approach but to add a builder for creating Pods/Containers similarly to the other builders we have.